IanGRap/Squish
# Pause Menu

Contributors: Alfredo, Austin

Goals:
1) Implement a method of pausing the game during a match
2) Implement an easy way to return to character select, quit the game and restart the match

Important Files:
1) MatchManager.cs 
    - When a player presses start, stop time and open _that player's_ pause menu
    - Exposed functions for Restart Match, Quit Game and Return to Player Select
    - Exposed a new variable called IsPaused
2) UIPauseMenu.cs
    - Controls the pause menu UI elements when IsPaused is true.
    - Calls exposed functions from MatchManager.cs
3) UIPauseMenuInput.cs
    - Handles controller / keyboard input logic for the pause menu
4) UIMatchManager.cs
    - Repurposed this file to generally encapsulate match UI instead of only Game Over text
5) UIGameOver.cs
    - Moved Game Over UI functionality from UIMatchManager.cs to this file

How to Test:
- IMPORTANT test in standalone build and Unity editor
- Load the scene testing/pause menu
- Press Start (escape on keyboard)
    - Make sure game time was paused
    - Make sure you can navigate the menu with up / down arrow keys, analog stick, dpad (inverted on PS4 controller)
    - Make sure that only the player who pressed pause can control the pause menu
    - Make sure you can exit menu with B and Start
    - Make sure that all menu options behave the way you expect when you press A (Right shift on keyboard)
        - Note: You can only Close Game if you are in standalone build
    - Make sure you hear sounds when you open the menu, close the menu and select an option

Notes: 
- We built our own manual hovering system instead of using Unity's built in navigation.
- In order to test, we changed the build settings to start the pause menu scene
